bin/yaml/cpp.yaml

```diff
@@ -804,6 +804,18 @@ compilers:
             chmod +x installer.bin
             ./installer.bin --unattendedmodeui none --mode unattended --prefix .
             rm installer.bin
+    intel-os-cpp:
+      type: script
+      dir: "intel-os-cpp-{name}"
+      check_exe: bin/clang++ --version
+      check_stderr_on_stdout: true
+      targets:
+        - name: 2023.0618
```

Comment:
As phrased this will once and once only install `2023.0618`.  We need to ensure this is tagged as `nightly` with

```yaml
if: nightly
```

somewhere, and we will need to consider how to do the nightly "rotation". There are two choices: either we `force: yes` or similar (I forget the exact syntax), which will unconditionally re-install "over the top" of the current build (probably ok), or else we need to have a `daily_script` `type:` which does the necessary care and feeding of the symlinks we use in other cases.